include/nuttx/input/sbutton.h

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************
+ * include/nuttx/input/sbutton.h
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/* The single button driver exports a standard character driver interface. By
+ * convention, the driver is exposed as /dev/sbtn[n] and works like a common
+ * keyboard device, but generates only INPUT_TAB and INPUT_ENTER events.
+ */
+
+#ifndef __INCLUDE_NUTTX_INPUT_SBUTTON_H
+#define __INCLUDE_NUTTX_INPUT_SBUTTON_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/i2c/i2c_master.h>
+#include <stdbool.h>
+#include <nuttx/irq.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* A reference to a structure of this type must be passed to the
+ * single button keyboard driver.  This structure provides information
+ * about the configuration and provides some board-specific hooks.
+ */
+
+struct sbutton_config_s
```

Comment:
why not write a general one key board driver on top of ioexpander(keyboard_ioexpander.c/keyboard_gpio.c)? so, we don't need add the duplicated glue logic for each board/chip?

Reply:
@xiaoxiang781216 although using ioexpander could bring more flexibility it makes the code more complex and also it could generate issues: a key from a real I2C ioexpander will not work very well with this driver.

Reply:
but why?